src/System.Private.CoreLib/shared/System/Diagnostics/StackTrace.cs

```diff
@@ -321,13 +321,6 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                             sb.AppendFormat(CultureInfo.InvariantCulture, inFileLineNum, fileName, sf.GetFileLineNumber());
                         }
                     }
-
-                    // Skip EDI boundary for async
-                    if (sf.IsLastFrameFromForeignExceptionStackTrace && !isAsync)
-                    {
-                        sb.AppendLine();
-                        sb.Append(SR.Exception_EndStackTraceFromPreviousThrow);
-                    }
```

Comment:
Why is this being removed?

Reply:
The idea was that the boundary could be inferred - this was in the previous PR I think.

Reply:
I don't believe that's easy to do in many/all cases.  We explicitly skip this for async because that case generally is easier, but for other cases, you're generally not nearly as linear, and you end up with stack traces that look impossible, which is much more confusing and time consuming to analyze than when they're clearly demarcated.